github-actions/runtime-variables/RuntimeVariables.MD

```diff
@@ -0,0 +1,234 @@
+
+```
+> git add.
+> git commit -m "Running Github Actions Test [commit var]"
+> git push
+```
+
+2. This triggers the workflow (as will any push). As the `[commit var]` is in the commit message, the `${COMMIT_VAR}` variable in the workflow will be set to `true` and result in the following:
+
+![Commit True Scenario](images/CommitTrue.PNG)
+
+##### Not Including the Variable
+
+
+1. Push to  branch `master`:
+
+```
+> git add.
+> git commit -m "Running Github Actions Test"
+> git push
+```
+
+2. This triggers the workflow (as will any push). As the `[commit var]` is **not** in the commit message, the `${COMMIT_VAR}` variable in the workflow will be set to `false` and result in the following:
+
+![Commit True Scenario](images/CommitFalse.PNG)
+
+
+### PR Body Variables
+
+When a PR is made, the PR Body can also be used to setup variables. These variables can be made available to all the workflow runs that stem from that PR, which can help ensure that commit messages are more informative and less cluttered, and reduces the work on the developer.
+
+Once again, this for an expected key and value. In this case, the key is `PR_VAR` and the value is `[pr var]`.
+
+#### Pre-requisites
+- Pipeline triggers are correctly set up to trigger on a pull request into a specific branch. (Here we will use master as the destination branch.)
+
+#### Code Snippet
+
+```
+on:
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  Echo-On-PR:
+    runs-on: ubuntu-latest
+    steps:
+      - name: "Checkout Repository"
+        uses: actions/checkout@v2
+
+      - name: "Set flag from PR"
+        env:
+          PR_VAR: ${{ contains(github.event.pull_request.body, '[pr var]') }}
+        run: |
+          if ${PR_VAR} == true; then
+            echo "::set-env name=flag::true"
+            echo "flag set to true"
+          else
+            echo "::set-env name=flag::false"
+            echo "flag set to false"
+          fi
+
+      - name: "Use flag if true"
+        if: env.flag
+        run: echo "Flag is available and true"
+```
+
+Available as a .YAML [here](examples/pr-example.yaml).
+
+#### Code Explanation
+The first part of the YAML file simply sets up the Pull Request Trigger. The majority of the following code is identical, so we will only explain the differences.
+
+```
+- name: "Set flag from PR"
+  env:
+    PR_VAR: ${{ contains(toJSON(github.event.pull_request.body), '[pr var]') }}
+```
+
+In this section, the `PR_VAR` environment variable is set to `true` or `false` depending on whether or not the `[pr var]` string is in the PR Body.
+
+#### Shorter Alternative
+
+Similarly to the above, the YAML step can be simplified to the following in order to make the code much shorter, although not necessarily more readable:
+
+```
+- name: "Set flag from PR"
+  env:
+    PR_VAR: ${{ contains(github.event.pull_request.body, '[pr var]') }}
+  run: |
+  echo "::set-env name=flag::${PR_VAR}"
+  echo "set flag to ${PR_VAR}"
+```
+
+#### Usage
```

Comment:
I might consider adding a section for real-world usage similar to how we used this concept in a previous project. What other things could developers consider doing with this? I.e. creating on-demand integration environments, setting runtime configs, etc

Reply:
Done!